src/Workspaces/CoreTestUtilities/Remote/TestSerializerService.cs

```diff
@@ -62,20 +58,17 @@ public override void WriteMetadataReferenceTo(MetadataReference reference, Objec
             }
         }
 
-        public override MetadataReference ReadMetadataReferenceFrom(ObjectReader reader, CancellationToken cancellationToken)
-        {
-            if (reader.ReadBoolean())
-            {
-                // this is a well-known reference
-                return s_wellKnownReferences[reader.ReadString()];
-            }
-            else
-            {
-                return base.ReadMetadataReferenceFrom(reader, cancellationToken);
-            }
-        }
+        protected override MetadataReference ReadMetadataReferenceFrom(ObjectReader reader, CancellationToken cancellationToken)
+            => reader.ReadBoolean()
+                ? s_wellKnownReferences[reader.ReadRequiredString()]
+                : base.ReadMetadataReferenceFrom(reader, cancellationToken);
+
+        protected override Checksum CreateChecksum(AnalyzerReference reference, CancellationToken cancellationToken)
+            => reference is TestGeneratorReference generatorReference
+                ? generatorReference.Checksum
+                : base.CreateChecksum(reference, cancellationToken);
```

Comment:
this is new, and replaced the existing hijacking point that was using IChecksummedObject.